File: src/main/java/com/blogly/blogly/infrastructure/persistence/user/UserDomainMapper.java
package com.blogly.blogly.infrastructure.persistence.user;

import com.blogly.blogly.domain.user.*;
import org.springframework.stereotype.Component;

@Component
public class UserDomainMapper {

    public UserEntity toEntity(User user) {
        UserEntity entity = new UserEntity();
        entity.setId(user.getId().getValue());
        entity.setEmail(user.getEmail().getValue());
        entity.setPassword(user.getPassword().getHashedValue());
        entity.setName(user.getName().getValue());
        entity.setRole(user.getRole());

        return entity;
    }

    public User toDomain(UserEntity entity) {
        return new User(
                new UserId(entity.getId()),
                new Email(entity.getEmail()),
                new Password(entity.getPassword()),
                new Name(entity.getName()),
                entity.getRole()
        );
    }
}

File: src/main/java/com/blogly/blogly/infrastructure/persistence/user/UserDomainMapper.kt
package com.blogly.blogly.infrastructure.persistence.user

import com.blogly.blogly.domain.user.*
import kotlin.time.toJavaInstant
import kotlin.time.toKotlinInstant

object UserDomainMapper {

    fun toEntity(user: User): UserEntity =
        UserEntity().apply {
            id = user.id.value
            email = user.email.value
            password = user.password.hashedValue
            name = user.name.value
            role = user.role
            bio = user.bio?.value
            avatarKey = user.avatarKey?.value
            joinedAt = user.joinedAt.toJavaInstant()
        }

    fun toDomain(entity: UserEntity): User =
        User(
            UserId(entity.id),
            Email(entity.email),
            Password(entity.password),
            Name(entity.name),
            entity.role,
            entity.bio?.let(::Bio),
            entity.avatarKey?.let(::AvatarKey),
            entity.joinedAt.toKotlinInstant(),
        )
}
